src/lib/mode.ts

```diff
@@ -1,145 +1,110 @@
 // Modified version of the light switch by: https://skeleton.dev
 
-import { derived, get } from 'svelte/store';
 import { persisted } from 'svelte-persisted-store';
 
 /**
  * Stores
- * ----
- * TRUE: Light mode | FALSE: Dark mode
  */
 
-const modeOsPrefers = persisted<boolean>('modeOsPrefers', false);
-const modeUserPrefers = persisted<boolean | undefined>('modeUserPrefers', undefined);
-const modeCurrent = persisted<boolean>('modeCurrent', false);
-
-/** Derived store with either `"light"` or `"dark"` depending on the current mode */
-export const mode = derived(modeCurrent, ($modeCurrent) => ($modeCurrent ? 'light' : 'dark'));
+const systemPrefersMode = persisted<'dark' | 'light'>('systemPrefersMode', 'dark');
+const userPrefersMode = persisted<'dark' | 'light' | undefined>('userPrefersMode', undefined);
+export const mode = persisted<'dark' | 'light'>('mode', 'dark');
```

Comment:
I think we should avoid exporting the persisted store and instead export a derived store so someone doesn't get the impression that they can just do something like `$mode = 'light'` and have it update accordingly (which it won't due to needing to pass through the `setMode`.

By forcing mutations through the `setMode` function we can ensure the right classes/style attributes are applied.

```suggestion
const activeMode = persisted<'dark' | 'light'>('mode', 'dark');
export const mode = readonly(activeMode)
```

Reply:
> I think we should avoid exporting the persisted store and instead export a derived store so someone doesn't get the impression that they can just do something like `$mode = 'light'` and have it update accordingly (which it won't due to needing to pass through the `setMode`.
> 
> By forcing mutations through the `setMode` function we can ensure the right classes/style attributes are applied.

got it! makes sense